Sources/ComposableArchitecture/Store.swift

```diff
@@ -121,20 +121,53 @@ public final class Store<State, Action> {
   private let reducer: (inout State, Action) -> Effect<Action, Never>
   private var bufferedActions: [Action] = []
   #if DEBUG
-    private let initialThread = Thread.current
+  private let dispatchToken: UUID?
   #endif
 
-  /// Initializes a store from an initial state, a reducer, and an environment.
+  /// Initializes a store from an initial state, a reducer, an environment, and a dispatch queue.
   ///
   /// - Parameters:
   ///   - initialState: The state to start the application in.
   ///   - reducer: The reducer that powers the business logic of the application.
   ///   - environment: The environment of dependencies for the application.
-  public init<Environment>(
+  ///   - dispatchQueue: Some dispatch queue onto which the store is assumed to process
+  ///   operations. If no queue is specified, the store should run on the main queue.
+  ///
+  /// - Note: You are responsible for dispatching store's operations on the correct queue.
+  public convenience init<Environment>(
+    initialState: State,
+    reducer: Reducer<State, Action, Environment>,
+    environment: Environment,
+    dispatchQueue: DispatchQueue = .main
```

Comment:
What do you think about naming this `bindToDispatchQueue`?

Reply:
This is indeed possible. My initial implementation used a factory `.attachedTo(dispatchQueue: DispatchQueue, initialState:…`. I'm still not sure if a factory is preferable to the initializer in order to preserve leanness of code completion.
Going back to `bindTo`, I find it a little awkward to have an active action in an initializer (which is already bearing some action itself). I would have thought to use the passive `boundTo` instead. The other small issue is the collision with `Binding` which can be confusing for users discovering the library.
However, I'm not a native speaker, so I'm not in the best position to evaluate this. What do you think?

Reply:
I'm fine with either `dispatchQueue` or `boundTo`.

Reply:
I went with boundTo with the last commit, as there are many references to the store being bound to some queue in the breakpoint message now.